tools/db/phone-users/src/PhoneUsers/Lib.hs

```diff
@@ -0,0 +1,172 @@
+import Wire.API.User (AccountStatus (Active))
+
+lookupClientsLastActiveTimestamps :: ClientState -> UserId -> IO [Maybe UTCTime]
+lookupClientsLastActiveTimestamps client u = do
+  runClient client $ runIdentity <$$> retry x1 (query selectClients (params LocalQuorum (Identity u)))
+  where
+    selectClients :: PrepQuery R (Identity UserId) (Identity (Maybe UTCTime))
+    selectClients = "SELECT last_active from clients where user = ?"
+
+readUsers :: ClientState -> ConduitM () [UserRow] IO ()
+readUsers client =
+  transPipe (runClient client) (paginateC selectUsersAll (paramsP LocalQuorum () 200) x5)
+    .| Conduit.map (fmap CQL.asRecord)
+  where
+    selectUsersAll :: C.PrepQuery C.R () (CQL.TupleType UserRow)
+    selectUsersAll =
+      "SELECT id, email, phone, activated, status, team FROM user"
+
+getConferenceCalling :: ClientState -> TeamId -> IO (Maybe FeatureStatus)
+getConferenceCalling client tid = do
+  runClient client $ runIdentity <$$> retry x1 (query1 select (params LocalQuorum (Identity tid)))
+  where
+    select :: PrepQuery R (Identity TeamId) (Identity FeatureStatus)
+    select =
+      "select conference_calling from team_features where team_id = ?"
+
+process :: Maybe Int -> ClientState -> ClientState -> IO Result
+process limit brigClient galleyClient =
+  runConduit $
+    readUsers brigClient
+      .| Conduit.mapM (\chunk -> SIO.hPutStr stderr "." $> chunk)
+      .| Conduit.concat
+      .| (maybe (Conduit.filter (const True)) Conduit.take limit)
+      .| Conduit.mapM (getUserInfo brigClient galleyClient)
+      .| Conduit.foldMap infoToResult
+
+getUserInfo :: ClientState -> ClientState -> UserRow -> IO UserInfo
+getUserInfo brigClient galleyClient ur = do
+  if not $ isCandidate
+    then pure NoPhoneUser
+    else do
+      -- should we give C* a little break here and add a small threadDelay?
+      threadDelay 200
+      lastActiveTimeStamps <- lookupClientsLastActiveTimestamps brigClient ur.id
+      now <- getCurrentTime
+      -- activity:
+      --   inactive: they have no client or client's last_active is greater than 90 days ago
+      --   active: otherwise
+      -- last_active is null on client creation, but it will be set once notifications are fetched
+      -- therefore we can consider empty last_active as inactive
+      let activeLast90Days = any (clientWasActiveLast90Days now) $ catMaybes lastActiveTimeStamps
+      userInfo <-
+        if activeLast90Days
+          then do
+            apu <- case ur.team of
+              Nothing -> pure ActivePersonalUser
+              Just tid -> do
+                isPaying <- isPayingTeam galleyClient tid
+                pure $
+                  if isPaying
+                    then ActiveTeamUser Free
+                    else ActiveTeamUser Paid
+            putStrLn ""
+            putStrLn $ "active phone user found: " <> show apu
+            print ur
+            putStrLn $ "last active time stamps: " <> show lastActiveTimeStamps
```

Comment:
Why not use the logger that was created? This way things can even be printed as JSON then we can easily answer more questions about the users than just the final `Result`.